File: problems/0009_palindrome-number/to-string/Solution.kt
class Solution {
    fun isPalindrome(x: Int): Boolean {
        if (x < 0)
            return false
        else if (x < 10)
            return true
        val s = x.toString()
        var i = 0
        var j = s.length - 1
        while (i < j)
            if (s[i++] != s[j--])
                return false
        return true
    }
}

File: problems/0009_palindrome-number/to-string/Solution.cs
public class Solution {
    public bool IsPalindrome(int x) {
        if (x < 0)
            return false;
        else if (x < 10)
            return true;
        var s = x.ToString();
        for (int i = 0, j = s.Length - 1; i < j; ++i, --j)
            if (s[i] != s[j])
                return false;
        return true;
    }
}
